third-party/qthread/Makefile

```diff
@@ -15,33 +17,18 @@ QTHREAD_INSTALL_DIR = $(QTHREAD_ABS_DIR)/$(QTHREAD_INSTALL_SUBDIR)
 QTHREAD_BUILD_DIR = $(QTHREAD_ABS_DIR)/$(QTHREAD_BUILD_SUBDIR)
 QTHREAD_DIR = $(QTHREAD_ABS_DIR)
 
-#
-# Cray builds are cross-compilations
-#
-ifneq (, $(filter cray-x%,$(CHPL_MAKE_TARGET_PLATFORM)))
-CHPL_QTHREAD_CFG_OPTIONS += --host=$(CHPL_MAKE_TARGET_ARCH)-cle-linux-gnu
-else ifneq (, $(filter cray-cs hpe-cray-ex,$(CHPL_MAKE_TARGET_PLATFORM)))
-CHPL_QTHREAD_CFG_OPTIONS += --host=$(CHPL_MAKE_TARGET_ARCH)-unknown-linux-gnu
-endif
-
 ifneq ($(CHPL_MAKE_HWLOC),none)
```

Comment:
I don't see the equivalent new code for this with CMake

Reply:
I don't remember this one offhand. I'll revisit it and report back.

Reply:
Okay, I'm definitely not certain about this one since I'm not familiar with the build pipelines for these systems, but as far as I can tell there's nothing needed here. These arguments are just about making sure that `AC_CANONICAL_HOST` works correctly. This used to be much more important to the qthreads build system since it did a ton of configure-time checking. Prior to switching to CMake I also made it so that the qthreads headers get all of this information directly from the compiler's preprocessor instead so there's less needed here. Also, I don't see any equivalent of this cross-compiling logic anywhere else where the chapel build system interacts with cmake, so presumably CMake is already doing the right thing.

Reply:
I am not sure about it being a CMake thing versus autoconf thing, but the only other CMake thing to compare it to is LLVM (which is used on the host, not in the runtime). All of the other third-party runtime components have this logic, except for gasnet and re2.  But I see your point about it being something needed for configure time, which is now handled by qthread headers. So I am fine with this, and I'll keep an eye on the testing on those systems to make sure there is no problem

Reply:
Thanks, that sounds like a good plan. I think it'll work, but I have no idea how/where to test this.